src/utils/utils.js

```diff
@@ -3,3 +3,59 @@ export function isStateless(component) {
     // `() => {}` via Babel has prototype too.
     return !(component.prototype && component.prototype.render)
 }
+
+function getMixins(target, methodName) {
+    const mixins = (target["__$mobxMixins"] = target["__$mobxMixins"] || {})
+    const methodMixins = (mixins[methodName] = mixins[methodName] || {})
+    methodMixins.pre = methodMixins.pre || []
+    methodMixins.post = methodMixins.post || []
+    return methodMixins
+}
+
+export function patch(target, methodName, mixinMethod, runMixinFirst = false) {
```

Comment:
could you elaborate why this enhanced patch was needed?

Reply:
for two reasons. 
the first one is that the patching method would only support one decorator override, so when this decorator was being used with observer one of the two would run, but not both.
the second one is that the current patch method works OK for prototype methods, but not for property methods defined as arrow function, so this new patch method supports both

Reply:
Quoting the PR description:
Also since I was getting weird interactions between @observer and this decorator (since both try to tweak componentWillUnmount) I had to adjust the way of patching lifecycle methods. The good thing is that I think it is more robust (should work with native classes, works with lifecycle methods set as arrow functions, is able to compose "mixins", etc)